lib/connection.cpp

```diff
@@ -1745,6 +1745,15 @@ QStringList Connection::stableRoomVersions() const
     return l;
 }
 
+bool Connection::canChangePassword() const
+{
+    if (!d->capabilities.changePassword) {
+        // assume we can
+        return true;
+    }
+    return d->capabilities.changePassword->enabled;
```

Comment:
Slightly more idiomatic (too bad that `value_or()` is useless here):
```suggestion
    // By default assume we can
    return d->capabilities.changePassword
           ? d->capabilities.changePassword->enabled
           : true;
```